lib/urldata.h

```diff
@@ -855,6 +855,7 @@ struct Curl_handler {
 #define PROTOPT_STREAM (1<<9) /* a protocol with individual logical streams */
 #define PROTOPT_URLOPTIONS (1<<10) /* allow options part in the userinfo field
                                       of the URL */
+#define PROTOPT_HTTP_PROXY (1<<11) /* allow over a HTTP proxy */
```

Comment:
why don't we just put it directly in the flag, PROTOPT_ALLOW_HTTP_PROXY or something